js/navigation.js

```diff
@@ -44,7 +44,18 @@
       toPop.classList.add('next');
       stack[stack.length - 1].classList.remove('back');
 
-      return this.waitForTransition();
+      return this.waitForTransition().then(() => {
+        toPop.dispatchEvent(new CustomEvent('pop'));
+      });
+    },
+
+    popToRoot: function() {
+      return this.pop().then(() => {
+        if (stack.length == 1) {
+          return;
+        }
+        return this.pop();
+      });
```

Comment:
this is really hacky; it doesn't work in 1 existing case and 1 possible future case:

existing case: when the app is in the main panel, we're popping where we should not. This happens when the app is launched and in the main panel -- and when the app is not launched (because we launch it in the main panel).
possible future case: obviously if our depth can be more than 2.

I'd do something like:

I don't think we really need to wait for each pop so I'd do something like this (untested):
```javascript
popToRoot: function() {
  var currentLength = stack.length;
  var pops = [];
  for (var i = 0; i < currentLength - 1; i++) {
    pops.push(this.pop());
  }
  return Promise.all(pops).then(() => {});
}
```